### PR TITLE
Add AI guidelines to a CLAUDE.md file

### PR DESCRIPTION
**Pull request by gautierronan**
See the [PyTorch CLAUDE.md](https://raw.githubusercontent.com/pytorch/pytorch/refs/heads/main/CLAUDE.md) for a good reference.

### CODE REVIEW

File: CLAUDE.md [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+`task doctest-docs`, `task ci` and `task docbuild` are CI's job, unless the user asks.
+Run the narrowest relevant target instead:
+
+```shell
+uv run pytest tests/sesolve/test_adaptive.py -q     # one file
+uv run pytest tests/mesolve -q -k gradient          # one concern
+uv run pytest dynamiqs/utils/general.py -q          # one module's docstring examples
+```
+
+## Repository layout
```
Comment (Nicolas-Lepage) [inline CLAUDE.md:43]: what do you think of adding this : 

Solver prefixes: se/me = Schrödinger / master equation; j/d = jump / diffusive
unravelling. So jsme = jump stochastic master equation.

File: CLAUDE.md [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+
+- Correctness is checked **against analytical solutions only** — never another solver, a
+  stored baseline, or QuTiP. The closed-form systems live in `tests/systems/`.
+- Subclass `IntegratorTester` and delegate to `_test_correctness`, `_test_gradient`,
+  `_test_hessian`; pass tolerances and solver options as keyword arguments rather than
+  loosening defaults.
+- Parametrize over both layouts (`dense_cavity` **and** `dia_cavity`) wherever dense and
+  sparse-DIA code paths can differ.
+- Keep the suite fast: existing files deliberately test only `Tsit5()` among the adaptive
+  methods. Do not parametrize over every available method.
```
Comment (Nicolas-Lepage) [inline CLAUDE.md:129]: should we add a general comment for all tests that is : 

if the change requires relaxing an assertion, changing a tolerance, or disabling a test to pass, stop and report instead of proceeding

File: CLAUDE.md [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+- Shapes use `...` for batch dimensions and `n` for the Hilbert dimension.
+- `[None, ...]` to add an axis, not `unsqueeze`; `(...).sum(0)` over `jnp.sum(..., 0)`.
+
+### Guidelines
+
+- **Minimize comments.** A comment earns its place by giving context that is not visible
+  locally — the physics being implemented, the reference an equation comes from, or why a
+  numerically obvious formulation was rejected. Never restate the syntax. Unicode math in
+  comments is idiomatic here (`# rho = Σ_i r_i |r_i⟩⟨r_i|`); in docstrings, math goes in
+  LaTeX instead, because it is rendered.
```
Comment (Nicolas-Lepage) [inline CLAUDE.md:189]: Agents tend to write comments that carry the reasoning of the session that produced the code

what do you think about adding something like : 

> **Comments describe the code as it stands, not how it got there.** The reader has
  no access to the conversation or the previous version — never "as requested", "now
  uses", "the previous approach". A rejected alternative earns a comment only when the
  reason is durable (`# lax.cond is not reverse-mode differentiable through the
  predicate`), never because it was tried today. Same for docstrings and identifiers:
  no `_new`, `_v2`, `_fixed`. The reasoning that led here belongs in the commit message.

**Review by Nicolas-Lepage — COMMENTED** [review]
The claude.md is seems very good and i am eager to test it ! 

i added a few ideas but do not know if they are useful.

File: CLAUDE.md [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+  degenerate parameters — `closed_system.py` picks `t_end=0.3` rather than a full period
+  specifically to avoid null gradients, which silently pass any gradient test.
+- `pytest.mark.parametrize` over loops, with readable `ids=` when not self-describing.
+- Random inputs always seeded via `dq.random.*` with an explicit key.
+- `pytest.raises(SomeError, match=...)` must assert the exact type — the distinction
+  between `NotImplementedError`, `TypeError` and `ValueError` is deliberate here.
+- A bug fix ships with a regression test that fails before the fix.
+- Docstring examples are tests too: changing an example's output breaks CI.
+
+## Coding style
```
Comment (Nicolas-Lepage) [inline CLAUDE.md:169]: to make @abocquet happy, restricting abbreviations for object names or variables could be a good guideline.

- **No abbreviations.** Full English words. Exception: symbols already standard in the papers/docstrings
  (`n`, `...`).

**Comment by gautierronan** [comment]
Agreed with all your comments! Made the appropriate changes in the last commit

**Review by Nicolas-Lepage — APPROVED** [review]
Thanks for the modifications, I am eager to see how this helps in writing code ! 
I also think both that versioning this will be useful to distill what works
